### ISSUE TITLE
Unable to connect to internet and update channels upon creating NixOS WSL2 instance 

### ISSUE THREAD

**Issue by RyzeNGrind** [nix-community/NixOS-WSL#409]
## Bug description

I tried to update the nix channels as instructed, but was unable to connect to internet to update.
## To Reproduce
Install NixOS-WSL and open a new nixOS shell. 
Read welcome message and then try to update with `sudo nix-channel --update`

## Attempts to resolve
I tried to add google and cloudflare dns servers manually, but still no resolution.
## Logs
```
[nixos@nixos:~]$ nixos-wsl-welcome
Welcome to your new NixOS-WSL system!

Please run `sudo nix-channel --update` and `sudo nixos-rebuild switch` now, to ensure you're running the latest NixOS and NixOS-WSL versions.

If you run into issues, please report them on our Github page at https://github.com/nix-community/NixOS-WSL or come talk to us on Matrix at #wsl:nixos.org.

❄️ Enjoy NixOS-WSL! ❄️

Note: this message will disappear after you rebuild your system. If you want to see it again, run `nixos-wsl-welcome`.

[nixos@nixos:~]$ sudo nix-channel --update
warning: error: unable to download 'https://nixos.org/channels/nixos-23.11': Couldn't resolve host name (6); retrying in 290 ms
warning: error: unable to download 'https://nixos.org/channels/nixos-23.11': Couldn't resolve host name (6); retrying in 585 ms
warning: error: unable to download 'https://nixos.org/channels/nixos-23.11': Couldn't resolve host name (6); retrying in 1002 ms
warning: error: unable to download 'https://nixos.org/channels/nixos-23.11': Couldn't resolve host name (6); retrying in 2308 ms
error: unable to download 'https://nixos.org/channels/nixos-23.11': Couldn't resolve host name (6)

[nixos@nixos:~]$ cat /etc/resolv.conf
# This file was automatically generated by WSL. To stop automatic generation of this file, add the following entry to /etc/wsl.conf:
# [network]
# generateResolvConf = false
#nameserver 172.22.80.1
nameserver 1.1.1.1
nameserver 8.8.8.8

[nixos@nixos:~]$ sudo cat /etc/wsl.conf
[automount]
enabled=true
mountFsTab=false
options=metadata,uid=1000,gid=100
root=/mnt

[boot]
command=
systemd=true

[interop]
appendWindowsPath=true
enabled=true

[network]
generateHosts=true
generateResolvConf=true
hostname=nixos

[user]
default=nixos

[nixos@nixos:~]$ ping -c 2 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.

--- 1.1.1.1 ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 1079ms


[nixos@nixos:~]$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1280 qdisc mq state UP group default qlen 1000
    link/ether ...{REDACTED}... brd ff:ff:ff:ff:ff:ff
    inet 172.22.95.125/20 brd 172.22.95.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 ...{REDACTED}.../64 scope link
       valid_lft forever preferred_lft forever

[nixos@nixos:~]$ cat /etc/resolv.conf
# This file was automatically generated by WSL. To stop automatic generation of this file, add the following entry to /etc/wsl.conf:
# [network]
# generateResolvConf = false
#nameserver 172.22.80.1
nameserver 1.1.1.1
nameserver 8.8.8.8

[nixos@nixos:~]$ ping -c 2 google.com
ping: google.com: Temporary failure in name resolution
```

## WSL version

```
C:\Users\{REDACTED_USER}>wsl -l -v
  NAME                   STATE           VERSION
* NixOS                  Running         2
  Ubuntu-22.04           Stopped         2
  docker-desktop-data    Stopped         2
  docker-desktop         Stopped         2

C:\Users\{REDACTED_USER}>wsl -v
WSL version: 2.0.9.0
Kernel version: 5.15.133.1-1
WSLg version: 1.0.59
MSRDC version: 1.2.4677
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.3996
```

**Comment by nzbr** [comment]
Are you able to access the internet from another WSL distro? As you can't ping `1.1.1.1` it's likely an issue with WSL, not NixOS, because it shouldn't usually touch the network interface setup

**Comment by RyzeNGrind** [comment]
> Are you able to access the internet from another WSL distro? As you can't ping `1.1.1.1` it's likely an issue with WSL, not NixOS, because it shouldn't usually touch the network interface setup

I have checked for connectivity in my Ubuntu WSL distro and my NixOS WSL distro and they both seem to have the same issue. I have followed standard instructions on the WSL and net re: setting `.wsl` and `.wslconfig` config files and haven't been able to find any resolution.

**Comment by SuperSandro2000** [comment]
Sorry but we cannot offer generic WSL support. If it is also broken in Ubuntu then this is not a NixOS WSL specific issue. Please seek the wider Internet for more help.

**Comment by justinrubek** [comment]
I am experiencing a similar issue, except other distros in WSL do work. I think it might be something funky with the network I'm on because I've had better luck using my home network in the past. When I go into my work's office, I can no longer use NixOS in WSL, but Ubuntu works out of the box.

Can anyone give me guidance on how I might troubleshoot this and what I should be looking for? I can't ping any addresses from the NixOS side.

**Comment by nikolarobottesla** [comment]
fix WSL networking in corporate environments: https://gist.github.com/machuu/7663aa653828d81efbc2aaad6e3b1431

TLDR:
```bash
sudo bash -c 'echo -e "[network]
generateResolvConf = false" > /etc/wsl.conf
rm /etc/resolv.conf
echo -e "options timeout:1 attempts:1 rotate
nameserver 1.1.1.1
nameserver 1.0.0.1" > /etc/resolv.conf
chattr -f +i /etc/resolv.conf'
#From <https://github.com/microsoft/WSL/issues/5420> 
```

**Comment by justinrubek** [comment]
I'm pretty skeptical of this solution because it isn't a DNS issue: my `resolv.conf` already has cloudflare's nameservers specified. 

Instead, it is the complete inability to ping any external server when using internet from that location. This is by IP, not domain name. I can take the same device back home and everything works fine, so it is definitely something with this specific network. That being said, it only affects NixOS-WSL and there is not much willpower from the organization to dig into this issue with me as a result.

I understand this is a volunteer community that can't devote significant resources to solving my work problems, but it seems there's some issue with the networking in NixOS-WSL that fails in this setup. I consider myself fairly savvy, but I'm not sure where to look to troubleshoot it, just that I know it isn't DNS.

**Comment by nzbr** [comment]
@justinrubek you could try setting `networking.firewall.enable = false;` and see if that helps. Besides that, I don't really see a way in which NixOS touches the network configuration that could result in it not working, while other distros do. If that doesn't help, try cheking for diffrences in the outputs of `ip a` and `ip r` between NixOS and a working distro
